x/delegation/keeper/delegation.go

```diff
@@ -62,6 +63,13 @@ func (k *Keeper) delegateTo(
 		if err := k.bankKeeper.DelegateCoinsFromAccountToModule(ctx, params.StakerAddress, delegationtype.DelegatedPoolName, coins); err != nil {
 			return err
 		}
+		// auto associate it, if there is a match. note that both are byte versions of bech32 AccAddress.
+		if bytes.Equal(params.StakerAddress, params.OperatorAddress[:]) {
```

Comment:
do we still need the associate/dissociate function in delegation precompile in this case? since the evm address that initiates the client chain delegation request could share the same bytes representation with the operator address, and we can require that operator should initiate delegations request from client chain with an evm address sharing the same bytes representation with its exocore bech32 operator address(meaning the same private key)

Reply:
I thought the intention for the precompile's association function is to support it on non EVM chains